__tests__/user/user.query.test.js

```diff
@@ -582,3 +586,166 @@ describe('Testing obtaining information restrictions', () => {
+              }
+            }
+          `,
+          variables: {
+            id: user._id,
+          },
+          context: {
+            headers: {
+              token: adminToken,
+            },
+          },
+        });
+      }
+    });
+  });
+
+  test('should sort by name from a to z', async () => {
+    const compareResult = testUsersSet.map(user => user.firstName).sort();
+
+    const users = await getAllUsers(adminToken, { name: 1 }).then(res =>
+      chooseOnlyUsers(res)
+    );
+
+    expect(users).toBeDefined();
+    expect(users.map(user => user.firstName)).toEqual(compareResult);
+  });
+
+  test('should sort by name from z to a', async () => {
+    const compareResult = testUsersSet
+      .map(user => user.firstName)
+      .sort()
+      .reverse();
+
+    const users = await getAllUsers(adminToken, { name: -1 }).then(res =>
+      chooseOnlyUsers(res)
+    );
+
+    expect(users).toBeDefined();
+    expect(users.map(user => user.firstName)).toEqual(compareResult);
+  });
+
+  test('should sort by email from a to z', async () => {
+    const compareResult = testUsersSet.map(user => user.email).sort();
+
+    const users = await getAllUsers(adminToken, { email: 1 }).then(res =>
+      chooseOnlyUsers(res)
+    );
+
+    expect(users).toBeDefined();
+    expect(users.map(user => user.email)).toEqual(compareResult);
+  });
+
+  test('should sort by emaill from z to a', async () => {
+    const compareResult = testUsersSet
+      .map(user => user.email)
+      .sort()
+      .reverse();
+
+    const users = await getAllUsers(adminToken, { email: -1 }).then(res =>
+      chooseOnlyUsers(res)
+    );
+
+    expect(users).toBeDefined();
+    expect(users.map(user => user.email)).toEqual(compareResult);
+  });
+
+  test('should show only banned users', async () => {
+    const compareResult = testUsersSet
+      .filter(user => user.banned)
+      .map(user => ({ firstName: user.firstName, banned: user.banned }));
+
+    const users = await getAllUsers(adminToken, {}, { banned: true }).then(
+      res =>
+        chooseOnlyUsers(res).map(user => ({
+          firstName: user.firstName,
+          banned: user.banned,
+        }))
+    );
+
+    expect(users).toBeDefined();
+    users.forEach(user => {
+      expect(user).toEqual(
+        compareResult.filter(el => el.firstName === user.firstName)[0]
+      );
+    });
+  });
+
+  test('should show only not banned users', async () => {
+    const compareResult = testUsersSet
+      .filter(user => !user.banned)
+      .map(user => ({ firstName: user.firstName, banned: user.banned }));
+
+    const users = await getAllUsers(adminToken, {}, { banned: false }).then(
+      res =>
+        chooseOnlyUsers(res).map(user => ({
+          firstName: user.firstName,
+          banned: user.banned,
+        }))
+    );
+
+    expect(users).toBeDefined();
+    users.forEach(user => {
+      expect(user).toEqual(
+        compareResult.filter(el => el.firstName === user.firstName)[0]
```

Comment:
Use **find** method instead of **filter**

Reply:
thanks, good point